### PR TITLE
Add AV1 codec level checking for decoding

### PR DESCRIPTION
**Pull request by wangyan42164**
Signed-off-by: Yan Wang <yan.wang@linux.intel.com>

### CODE REVIEW

File: api/include/mfxstructures.h [hunk 1]
```diff
@@ -507,6 +507,22 @@ enum {
     MFX_PROFILE_AV1_MAIN                    = 1,
     MFX_PROFILE_AV1_HIGH                    = 2,
     MFX_PROFILE_AV1_PRO                     = 3,
+
+    MFX_LEVEL_AV1_2                         = 0,
```
Comment (dmitry-gurulev) [inline api/include/mfxstructures.h:511]: Value '0' is reserved for `MFX_LEVEL_UNKNOWN`. Please do mapping of public API's levels definition to specific `seq_level_idx` internally

File: _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp [hunk 2]
```diff
@@ -450,6 +450,35 @@ mfxStatus VideoDECODEAV1::Query(VideoCORE* core, mfxVideoParam* in, mfxVideoPara
 #endif
     }
 
+    if (in)
+    {
+        switch(in->mfx.CodecLevel)
```
Comment (mgonchar) [inline _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp:455]: can we wrap checking to separate helper function? Just to not expand code of the Query

File: api/include/mfxstructures.h (hunk 1 above)
Comment (mgonchar) [reply api/include/mfxstructures.h:511]: How about doing it similarly to HEVC in this enum, i.e.:
```
MFX_LEVEL_AV1_2                         = 20,
MFX_LEVEL_AV1_21                       = 21,
...
```

Comment (wangyan42164) [reply api/include/mfxstructures.h]: Sure. I have changed it in the latest commit.

File: _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp (hunk 2 above)
Comment (wangyan42164) [reply _studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp:438]: Sure. I have changed it in the latest commit.

**Comment by dmitryermilov** [comment]
@dmitry-gurulev , are there remaining concerns from you?

**Pull request merged**